kuma/static/js/newsletter.js

```diff
@@ -0,0 +1,221 @@
+                for (var i = 0; i < errorArray.length; i++) {
+                    var item = doc.createElement('li');
+                    item.appendChild(doc.createTextNode(errorArray[i]));
+                    errorList.appendChild(item);
+                }
+            } else {
+                // no error messages, forward to server for better troubleshooting
+                newsletterForm.setAttribute('data-skip-xhr', true);
+                newsletterForm.submit();
+            }
+
+            newsletterErrors.appendChild(errorList);
+            newsletterErrors.style.display = 'block';
+
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'error'
+            });
+        }
+
+        // show sucess message
+        function newsletterThanks() {
+            var thanks = doc.getElementById('newsletterThanks');
+
+            // show thanks message
+            thanks.style.display = 'block';
+
+            // track success
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'complete'
+            });
+
+            // don't show signup form on article pages anymore
+            newsletterSaveHide();
+        }
+
+        // XHR subscribe; handle errors; display thanks message on success.
+        function newsletterSubscribe(event) {
+            var skipXHR = newsletterForm.getAttribute('data-skip-xhr');
+            if (skipXHR) {
+                // track this
+                mdn.analytics.trackEvent({
+                    'category': 'newsletter',
+                    'action': 'progression',
+                    'label': 'error-forward'
+                });
+
+                return true;
+            }
+            event.preventDefault();
+            event.stopPropagation();
+
+            // track this submission
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'submission'
+            });
+
+            // new submission, clear old errors
+            errorArray = [];
+            newsletterErrors.style.display = 'none';
+            while (newsletterErrors.firstChild) newsletterErrors.removeChild(newsletterErrors.firstChild);
+
+            var fmt = doc.getElementById('fmt').value;
+            var email = newsletterEmailInput.value;
+            var newsletter = doc.getElementById('newsletterNewslettersInput').value;
+            var privacy = doc.querySelector('input[name="privacy"]:checked') ? '&privacy=true' : '';
+            var params = 'email=' + encodeURIComponent(email) +
+                         '&newsletters=' + newsletter +
+                         privacy +
+                         '&fmt=' + fmt +
+                         '&source_url=' + encodeURIComponent(doc.location.href);
```

Comment:
current method:
email=success%40example.com&newsletters=app-dev&privacy=true&fmt=H&source_url=https%3A%2F%2Fdeveloper-local.allizom.org%2Fen-US%2F

encodeURI(params):
email=success@example.com&newsletters=app-dev&privacy=true&fmt=H&source_url=https://developer-local.allizom.org/en-US/ 

I don't think encodeURI is smart enough to know the difference between a URL and a URL in a URL parameter :/